string[] source_product
string[] target_ms
string transition
---
